### PR TITLE
Login through GitHub using omniauth

### PR DESCRIPTION
**Pull request by fabio-guerreiro**

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -15,5 +17,10 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+
+    config.middleware.use OmniAuth::Builder do
+      provider :developer if Rails.env.development?
+      provider :github, '1d326984ab021aa71ac0', '2852cc2957f48ebb31095566d667ff7949589df0'
```
Comment (fwuensche) [inline config/application.rb:23]: never ever commit your access tokens. you can use rails credentials for that. 

```
EDITOR=vim rails credentials:edit
```

and please don't forget to delete these access tokens from your github account :P

File: config/application.rb [hunk 2]
```diff
@@ -15,5 +17,10 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+
+    config.middleware.use OmniAuth::Builder do
+      provider :developer if Rails.env.development?
```
Comment (fwuensche) [inline config/application.rb:22]: let's just use different access tokens for different environment and keep github as only provider, no?

File: config/application.rb (hunk 1 above)
Comment (fwuensche) [reply config/application.rb:23]: you'll need the secret key base from here https://dashboard.heroku.com/apps/uniq-production/settings

File: config/application.rb (hunk 2 above)
Comment (fabio-guerreiro) [reply config/application.rb:22]: Sure!

File: config/application.rb (hunk 1 above)
Comment (fabio-guerreiro) [reply config/application.rb:23]: Can't access it 😕

Comment (coisnepe) [reply config/application.rb]: I use `EDITOR="code --wait" rails credentials:edit`, which saves the credentials as soon as you close the tab

**Pull request merged**